feature_engine/creation/distance_features.py

```diff
@@ -0,0 +1,118 @@
+from typing import Optional, Union
+
+import numpy as np
+import pandas as pd
+
+from feature_engine._base_transformers.base_numerical import BaseNumericalTransformer
+from feature_engine._base_transformers.mixins import (
+    FitFromDictMixin,
+    GetFeatureNamesOutMixin,
+)
+from feature_engine._check_input_parameters.check_init_input_params import (
+    _check_param_drop_original,
+)
+
+
+class DistanceFeatures(
+    BaseNumericalTransformer, FitFromDictMixin, GetFeatureNamesOutMixin
+):
+    EARTH_RADIUS: float = 6371.  # radius of Earth in kms
+
+    def __init__(
+            self,
+            a_latitude: str,
```

Comment:
We allow integers as column names.

To think: is this transformer intended to compare the distance between 2 points?

Because what we normally do (see for example RelativeFeatures), is that we allow a list of variables in the first parameter, that would be we allow a list in a_latitude and a_longitude, and we allow for example a variable, or we could also think of allowing a list for b latitude and longitude, and then we calculate the distance between those in a and those in b.

This way, if the user wants to calculate more than 1 distance, they don't need to set multiple transformers.

What do you think?

For clarity, my df has for example, address victim 1, lat and lon, address victim 2, lat and lon, address perpetrator, lat and lon, and we want to calculate the distance between victim 1 and perpetrator and victim 2 and perpetrator. And if we had 2 perpetrators, then we want the distances of both victims to both perpetrators. Not sure why I am talking about victims and perpetrators, it sounds horrible. But it illustrates the issue :p

Reply:
I see your point. 

do you would like something as: `variables=[[lat_1, lon_1, lat_2, lon_2], [lat_1, lon_1, lat_3, lon_3]]` and the transformer compute the distance between lat_11, lon_1 and lat_2, lon_2 and the distance between lat_1, Lon_1 and lat_3, lon_3 ?

Reply:
If we have:

- variables=[lat_1, lon_1, lat_2, lon_2]
- reference=[lat_3, lon_3, lat_4, lon_4]

The result would be the following distances:

- lat_1;lon_1 and lat_3;lon3
- - lat_1;lon_1 and lat_4;lon4
- - lat_2;lon_2 and lat_3;lon3
- - lat_2;lon_2 and lat_3;lon3